src/BCEngineMCMC.cxx

```diff
@@ -2307,24 +2300,16 @@ void BCEngineMCMC::MCMCInitialize()
     // reset convergence
     fMCMCNIterationsConvergenceGlobal = -1;
 
-    // reset iteration counters
-    fMCMCNIterations.assign(fMCMCNChains, 0);
-
     // reset statistics counters
     fMCMCStatistics.assign(fMCMCNChains, BCEngineMCMC::Statistics(GetNParameters(), GetNObservables()));
     fMCMCStatistics_AllChains.Init(GetNParameters(), GetNObservables());
 
-    // reset likelihood & probability holders
-    fMCMCprob.assign(fMCMCNChains, -std::numeric_limits<double>::infinity());
-    fMCMCLogLikelihood.assign(fMCMCNChains, -std::numeric_limits<double>::infinity());
-    fMCMCLogLikelihood_Provisional.assign(fMCMCNChains, -std::numeric_limits<double>::infinity());
+    // reset states holder
+    fMCMCStates.assign(fMCMCNChains, ChainState(GetNObservables()));
```

Comment:
Why `GetNObservables()`. How are the number of parameters passed to `ChainState`?

Reply:
By assignment from `ThreadLocalStorage`. Only the `observables` vector needs to be given a size, since it is assigned into directly from a loop over the `ObservablesSet`.